packages/client/src/debounce.tsx

```diff
@@ -0,0 +1,18 @@
+export default function debounce(
+  func: Function,
+  wait: number,
+  immediate: boolean
+) {
+  let timeout: any
+  return function(this: any, ...args: any[]) {
+    const context = this
```

Comment:
If you use an arrow function for `later` then you wouldn't need the intermediate `context` variable.